robocop/checkers/comments.py

```diff
@@ -58,3 +58,34 @@ def check_comment_content(self, comment_token):
         if comment_token.value.startswith('#') and comment_token.value != '#':
             if not comment_token.value.startswith('# '):
                 self.report("missing-space-after-comment", lineno=comment_token.lineno, col=comment_token.col_offset)
+
+
+class IgnoredDataChecker(RawFileChecker):
+    """ Checker for ignored data. """
+    rules = {
+        "0704": (
+            "ignored-data",
+            "Ignored data found in file",
+            RuleSeverity.WARNING
+        )
+    }
+
+    def __init__(self, *args):
+        self.section_header = False
+        self.ignored_data = False
+        super().__init__(*args)
+
+    def parse_file(self):
+        with open(self.source) as file:
+            for lineno, line in enumerate(file, 1):
+                self.check_line(line, lineno)
```

Comment:
to reduce amount of code you can return boolean value from this function:

```
    def parse_file(self):
        with open(self.source) as file:
            for lineno, line in enumerate(file, 1):
                if self.check_line(line, lineno)
                    break

    def check_line(self, line, lineno):
        if line.startswith('***'):
            return True
        if not line.startswith('***') and not line.startswith('# robocop:'):
            self.report("ignored-data", lineno=lineno, col=0)
           return True
```
(init not needed in case). But code clarity will worsen a bit - your call

Reply:
Agree, I refactored the code.